src/main/java/com/puppycrawl/tools/checkstyle/checks/design/FinalClassCheck.java

```diff
@@ -253,14 +255,33 @@ public void leaveToken(DetailAST ast) {
      * @return true if given class should be declared as final otherwise false
      */
     private static boolean shouldBeDeclaredAsFinal(ClassDesc desc) {
-        return desc.isWithPrivateCtor()
+        return (desc.isWithPrivateCtor() || isPrivateWithDefaultCtor(desc))
                 && !(desc.isDeclaredAsAbstract()
                     || desc.isSuperClassOfAnonymousInnerClass())
                 && !desc.isDeclaredAsFinal()
                 && !desc.isWithNonPrivateCtor()
                 && !desc.isWithNestedSubclass();
     }
 
+    /**
+     * Check if class is private and with default constructor.
+     *
+     * @param desc description of class
+     * @return true if class is private and with default constructor
+     */
+    private static boolean isPrivateWithDefaultCtor(ClassDesc desc) {
```

Comment:
Where is the "Default CTOR" check? I only see a check for a private class. This seems to be missing as this is what the issue says.

> The default constructor has the same access modifier as the class.

If the ctor is defined and is private as the same as the class, it seems to me this should be triggered too.

Reply:
i have update the return statment please look

Reply:
Yes we are looking only for private class because all current code only look if private class is their with constructor using 
`desc.isWithPrivateCtor() && !desc.isWithNonPrivateCtor()` but what if their is no constructor if you don't find the naming good let me suggest something for that 

>  This seems to be missing as this is what the issue says.

I have put  `return (desc.isWithPrivateCtor() || isPrivateWithDefaultCtor(desc))` so if their is no constructor still will show the violation if expected

Reply:
> because all current code only look if private class is their with constructor

But this is not what I took the issue to be, which states we need to check the CTOR.

> I have put return (desc.isWithPrivateCtor() || isPrivateWithDefaultCtor(desc))

But this is outside `isPrivateWithDefaultCtor`. You named the method and made the javadoc say you are checking constructor.

Either `isPrivateWithDefaultCtor` is missing the CTOR check, or `isPrivateWithDefaultCtor` needs to remove wording of default ctor as there is no default check.

When you say "default ctor", I take that to mean the class has no constructor written. Your method only checks if the class given to it is private. It does not look for a "default ctor".

Reply:
@rnveach i have update the name and doc please look

Reply:
I still go back to
> But this is not what I took the issue to be, which states we need to check the CTOR.

Reply:
sorry but i dont understand `which states we need to check the CTOR`

Reply:
I think I am still lacking your point `which states we need to check the CTOR.`

--> current changes will return true in the method ` private static boolean shouldBeDeclaredAsFinal(ClassDesc desc) {`
if the class is private. 
--> If there is a constructor which is not default then it might be private or non private it will return true or false accordingly in ` private static boolean shouldBeDeclaredAsFinal(ClassDesc desc) {`.
--> if there is no constructor defined then it is case where there is default constructor if the class is private then according to https://docs.oracle.com/javase/specs/jls/se17/html/jls-8.html#jls-8.8.9 it is by default private. and it return true.

Reply:
can you please let me know what exactly is expected

Reply:
@rnveach ping